### ISSUE TITLE
Core dump on big regex

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6736]
Migrated from [rt.perl.org#23666](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=23666) (status was 'resolved')

Searchable as RT23666$

**Comment by p5pRT** [comment]
### From perl-5.8.0@ton.iguana.be

#### Created by perl-5.8.0@ton.iguana.be



perl \-wle '\("1"x5683\)=~ /^\(11\+\)\\1\+$/'

works ok

perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'
Segmentation fault \(core dumped\)

ulimit \-a
core file size \(blocks\)     unlimited
data seg size \(kbytes\)      unlimited
file size \(blocks\)          unlimited
max locked memory \(kbytes\)  unlimited
max memory size \(kbytes\)    unlimited
open files                  1024
pipe size \(512 bytes\)       8
stack size \(kbytes\)         1048576
cpu time \(seconds\)          unlimited
max user processes          6143
virtual memory \(kbytes\)     unlimited

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.8.0:

Configured by ton at Tue Nov 12 01:56:18 CET 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.19, archname=i686-linux-thread-multi-64int-ld
    uname='linux quasar 2.4.19 #5 wed oct 2 02:34:25 cest 2002 i686 unknown '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=define use64bitall=undef uselongdouble=define
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -fomit-frame-pointer',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.3 20010315 (release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='long double', nvsize=12, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -ldb -ldl -lm -lpthread -lc -lposix -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lpthread -lc -lposix -lcrypt -lutil
    libc=/lib/libc-2.2.4.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.2.4'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.0:
    /usr/lib/perl5/5.8.0/i686-linux-thread-multi-64int-ld
    /usr/lib/perl5/5.8.0
    /usr/lib/perl5/site_perl/5.8.0/i686-linux-thread-multi-64int-ld
    /usr/lib/perl5/site_perl/5.8.0
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.8.0:
    HOME=/home/ton
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/ton/bin.Linux:/home/ton/bin:/home/ton/bin.SampleSetup:/usr/local/bin:/usr/local/sbin:/usr/local/jre/bin:/home/oracle/product/9.0.1/bin:/usr/local/ar/bin:/usr/games/bin:/usr/X11R6/bin:/usr/share/bin:/usr/bin:/usr/sbin:/bin:/sbin:.
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @schwern

> \[perl\-5\.8\.0@&#8203;ton\.iguana\.be \- Sat Aug 30 14&#8203;:33&#8203;:01 2003\]&#8203;:
> 
> perl \-wle '\("1"x5683\)=~ /^\(11\+\)\\1\+$/'
> 
> works ok
> 
> perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'
> Segmentation fault \(core dumped\)
> 
> ulimit \-a
> core file size \(blocks\)     unlimited
> data seg size \(kbytes\)      unlimited
> file size \(blocks\)          unlimited
> max locked memory \(kbytes\)  unlimited
> max memory size \(kbytes\)    unlimited
> open files                  1024
> pipe size \(512 bytes\)       8
> stack size \(kbytes\)         1048576
> cpu time \(seconds\)          unlimited
> max user processes          6143
> virtual memory \(kbytes\)     unlimited

I can't reproduce with 5\.8\.0\, 5\.8\.6 or bleadperl on OS X\.  I can
reproduce on Debian 5\.8\.4 but I need to crank up the size a bit\.

$ perl \-wle '\("1"x 19973\)=~ /^\(11\+\)\\1\+$/'
Segmentation fault

I don't have a newer Perl on Linux to try out\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From shouldbedomo@mac.com

On 18/07/2005\, at 2&#8203;:05 AM\, Michael G Schwern via RT wrote&#8203;:

>> \[perl\-5\.8\.0@&#8203;ton\.iguana\.be \- Sat Aug 30 14&#8203;:33&#8203;:01 2003\]&#8203;:
>>
>> perl \-wle '\("1"x5683\)=~ /^\(11\+\)\\1\+$/'
>>
>> works ok
>>
>> perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'
>> Segmentation fault \(core dumped\)
>>
>> ulimit \-a
>> \.\.\.
>> stack size \(kbytes\)         1048576
>> \.\.\.
>>
>
> I can't reproduce with 5\.8\.0\, 5\.8\.6 or bleadperl on OS X\.  I can
> reproduce on Debian 5\.8\.4 but I need to crank up the size a bit\.
>
> $ perl \-wle '\("1"x 19973\)=~ /^\(11\+\)\\1\+$/'
> Segmentation fault
>
> I don't have a newer Perl on Linux to try out\.

Neither do I\. However\, I'll jump in anyway&#8203;:

Yet another case to be linked to the \#24274 deep\-regex\-recursion\- 
causes\-stack\-overflow metabug&#8203;:

$ perl \-v

This is perl\, v5\.8\.6 built for darwin\-thread\-multi\-2level

\.\.\.
$ ulimit \-a
core file size        \(blocks\, \-c\) 0
data seg size         \(kbytes\, \-d\) 6144
file size             \(blocks\, \-f\) unlimited
max locked memory     \(kbytes\, \-l\) unlimited
max memory size       \(kbytes\, \-m\) unlimited
open files                    \(\-n\) 256
pipe size          \(512 bytes\, \-p\) 1
stack size            \(kbytes\, \-s\) 8192
cpu time             \(seconds\, \-t\) unlimited
max user processes            \(\-u\) 100
virtual memory        \(kbytes\, \-v\) unlimited
$ ulimit \-s 2048
$ perl \-wle '\("1"x5683\)=~ /^\(11\+\)\\1\+$/'
$ ulimit \-s 1024
$ perl \-wle '\("1"x5683\)=~ /^\(11\+\)\\1\+$/'
Segmentation fault

 From the crash log&#8203;:

Host Name&#8203;:      Tullamore
Date/Time&#8203;:      2005\-07\-18 09&#8203;:45&#8203;:49\.115 \+0200
OS Version&#8203;:     10\.4\.2 \(Build 8C46\)
Report Version&#8203;: 3

Command&#8203;: perl
Path&#8203;:    /sw/bin/perl
Parent&#8203;:  bash \[22803\]

Version&#8203;: ??? \(???\)

PID&#8203;:    22826
Thread&#8203;: 0

Exception&#8203;:  EXC\_BAD\_ACCESS \(0x0001\)
Codes&#8203;:      KERN\_INVALID\_ADDRESS \(0x0001\) at 0xbfeffff0

Thread 0 Crashed&#8203;:
0   libSystem\.B\.dylib     0x90005e28 szone\_size \+ 28
1   libSystem\.B\.dylib     0x9001793c realloc \+ 1500
2   perl                  0x00003bec Perl\_safesysrealloc \+ 120 \(crt\.c&#8203;: 
300\)
3   perl                  0x00089020 Perl\_savestack\_grow\_cnt \+ 80  
\(crt\.c&#8203;:300\)
4   perl                  0x000b7664 S\_regcppush \+ 92 \(crt\.c&#8203;:300\)
5   perl                  0x000bdcc4 S\_regmatch \+ 8076 \(crt\.c&#8203;:300\)
6   perl                  0x000bdce0 S\_regmatch \+ 8104 \(crt\.c&#8203;:300\)
\.\.\.
507 perl                  0x000bdce0 S\_regmatch \+ 8104 \(crt\.c&#8203;:300\)
508 perl                  0x000bdce0 S\_regmatch \+ 8104 \(crt\.c&#8203;:300\)

Thread 0 crashed with PPC Thread State 64&#8203;:
&nbsp;   srr0&#8203;: 0x0000000090005e28 srr1&#8203;:  
0x100000000000d030                        vrsave&#8203;: 0x0000000000000000
&nbsp;     cr&#8203;: 0x48042224          xer&#8203;: 0x0000000020000000   lr&#8203;:  
0x0000000090005e1c  ctr&#8203;: 0x0000000090005e0c
&nbsp;     r0&#8203;: 0x000000009001793c   r1&#8203;: 0x00000000bff00040   r2&#8203;:  
0x00000000a0007374   r3&#8203;: 0x0000000001800000
&nbsp;     r4&#8203;: 0x0000000000205000   r5&#8203;: 0x0000000000000054   r6&#8203;:  
0x0000000000000000   r7&#8203;: 0x0000000000000031
&nbsp;     r8&#8203;: 0x0000000000000031   r9&#8203;: 0x0000000000000011  r10&#8203;:  
0x0000000000305e30  r11&#8203;: 0x00000000000e2828
&nbsp;    r12&#8203;: 0x0000000090005e0c  r13&#8203;: 0x0000000000000000  r14&#8203;:  
0x0000000000000000  r15&#8203;: 0x0000000000000000
&nbsp;    r16&#8203;: 0x0000000000000000  r17&#8203;: 0x0000000000000000  r18&#8203;:  
0x0000000000000000  r19&#8203;: 0x0000000000000000
&nbsp;    r20&#8203;: 0x0000000000000002  r21&#8203;: 0x0000000000000000  r22&#8203;:  
0x0000000000305be0  r23&#8203;: 0x00000000000109a4
&nbsp;    r24&#8203;: 0x0000000000205000  r25&#8203;: 0x000000000180e222  r26&#8203;:  
0x000000000180e224  r27&#8203;: 0x0000000000004263
&nbsp;    r28&#8203;: 0x0000000001800000  r29&#8203;: 0x0000000001800000  r30&#8203;:  
0x0000000000205000  r31&#8203;: 0x0000000090005e1c

Binary Images Description&#8203;:
&nbsp;     0x1000 \-    0xd9fff perl     /sw/bin/perl
0x8fe00000 \- 0x8fe51fff dyld 43\.1    /usr/lib/dyld
0x90000000 \- 0x901a6fff libSystem\.B\.dylib     /usr/lib/libSystem\.B\.dylib
0x901fe000 \- 0x90202fff libmathCommon\.A\.dylib     /usr/lib/system/ 
libmathCommon\.A\.dylib


Behaviour is the same with bleadperl@&#8203;25158\.
\-\- 
Dominic Dunlop

**Comment by p5pRT** [comment]
### From schubiger@cpan.org

> \[perl\-5\.8\.0@&#8203;ton\.iguana\.be \- Sat Aug 30 14&#8203;:33&#8203;:01 2003\]&#8203;:
> perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'
> Segmentation fault \(core dumped\)

Couldn't reproduce it with Perl 5\.8\.5 \- 5\.8\.7 nor Bleadperl 
on OpenBSD 3\.6\.

**Comment by p5pRT** [comment]
### From @smpeters

> \[perl\-5\.8\.0@&#8203;ton\.iguana\.be \- Sat Aug 30 14&#8203;:33&#8203;:01 2003\]&#8203;:
> 
> 
> This is a bug report for perl from perl\-5\.8\.0@&#8203;ton\.iguana\.be\,
> generated with the help of perlbug 1\.34 running under perl v5\.8\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> perl \-wle '\("1"x5683\)=~ /^\(11\+\)\\1\+$/'
> 
> works ok
> 
> perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'
> Segmentation fault \(core dumped\)
> 
> ulimit \-a
> core file size \(blocks\)     unlimited
> data seg size \(kbytes\)      unlimited
> file size \(blocks\)          unlimited
> max locked memory \(kbytes\)  unlimited
> max memory size \(kbytes\)    unlimited
> open files                  1024
> pipe size \(512 bytes\)       8
> stack size \(kbytes\)         1048576
> cpu time \(seconds\)          unlimited
> max user processes          6143
> virtual memory \(kbytes\)     unlimited
> 

This problem has been resolved with change \#27598\.

steve@&#8203;kirk&#8203;:\~/smoke/perl\-current$ perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'
Segmentation fault
steve@&#8203;kirk&#8203;:\~/smoke/perl\-current$ \./perl \-wle '\("1"x 9973\)=~ /^\(11\+\)\\1\+$/'

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
